android/src/main/kotlin/com/gdelataillade/alarm/features/AudioHandler.kt

```diff
@@ -25,52 +30,66 @@ class AudioHandler(private val context: Context) {
-                "flutter_assets/$filePath"
-            } else {
-                filePath
-            }
-
-        try {
-            MediaPlayer().apply {
-                if (adjustedFilePath.startsWith("flutter_assets/")) {
-                    // It's an asset file
-                    val assetManager = context.assets
-                    val descriptor = assetManager.openFd(adjustedFilePath)
-                    setDataSource(descriptor.fileDescriptor, descriptor.startOffset, descriptor.length)
+            val adjustedFilePath =
+                if (filePath.startsWith("assets/")) {
+                    "flutter_assets/$filePath"
                 } else {
-                    // Handle local files
-                    setDataSource(adjustedFilePath)
+                    filePath
                 }
 
-                prepare()
-                isLooping = loopAudio
-                start()
+            try {
+                MediaPlayer().apply {
+                    if (adjustedFilePath.startsWith("flutter_assets/")) {
+                        // It's an asset file
+                        val assetManager = context.assets
+                        val descriptor = assetManager.openFd(adjustedFilePath)
+                        setDataSource(descriptor.fileDescriptor, descriptor.startOffset, descriptor.length)
+                    } else {
+                        // Handle local files
+                        setDataSource(adjustedFilePath)
+                    }
+                    setWakeMode(context, PowerManager.PARTIAL_WAKE_LOCK)
+                    prepareAsync()
 
-                setOnCompletionListener {
-                    if (!loopAudio) {
-                        onAudioComplete?.invoke()
+                    setOnPreparedListener {
+                        isLooping = loopAudio
+                        start()
                     }
-                }
 
-                mediaPlayers[id] = this
+                    setOnCompletionListener {
+                        if (!loopAudio) {
+                            // Switch back to the main thread to update UI or state
+                            scope.launch(Dispatchers.Main) {
+                                onAudioComplete?.invoke()
+                            }
+                        }
+                    }
+
+                    mediaPlayers[id] = this
 
-                if (fadeDuration != null && fadeDuration > 0) {
-                    val timer = Timer(true)
-                    timers[id] = timer
-                    startFadeIn(this, fadeDuration, timer)
+                    if (fadeDuration != null && fadeDuration > 0) {
+                        // Timer operations should also be offloaded
+                        scope.launch(Dispatchers.Default) {
+                            val timer = Timer(true)
+                            timers[id] = timer
+                            startFadeIn(this@apply, fadeDuration, timer)
```

Comment:
If we are using coroutines, we can get rid of these timers. Just simple loop with delay inside suspend method.

Reply:
done